great_expectations/execution_engine/sqlalchemy_execution_engine.py

```diff
@@ -1192,13 +1199,23 @@ def _build_selectable_from_batch_spec(
                     )
                 )
 
-        return (
-            sa.select("*")
-            .select_from(
-                sa.table(table_name, schema=batch_spec.get("schema_name", None))
+        return sa.select("*").select_from(subselectable).where(split_clause)
+
+    def _subselectable(self, batch_spec: BatchSpec) -> Selectable:
+        table_name = batch_spec.get("table_name")
+        query = batch_spec.get("query")
+        selectable: Selectable
+        if table_name:
+            selectable = sa.table(
+                table_name, schema=batch_spec.get("schema_name", None)
             )
-            .where(split_clause)
-        )
+        else:
+            if not isinstance(query, str):
+                raise ValueError(f"SQL query should be a str but got {query}")
+            # Query is a valid SELECT query that begins with r"select\w"
+            selectable = sa.select(sa.text(query[6:].lstrip()))
```

Comment:
@billdirks Would it make sense to utilize some kind of an SQL query parser into tokens -- or something equivalent on our part -- to allow for a flexible number `\s`, `\t`, `\r`, `\n`, etc. characters in the query string?  Thanks.